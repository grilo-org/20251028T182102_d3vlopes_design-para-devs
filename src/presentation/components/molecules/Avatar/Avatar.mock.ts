import { AvatarProps } from '.';

export const avatarMock: AvatarProps = {
	src: 'https://media.graphassets.com/EDKFiM0SUS8flWAN2W7b',
	alt: 'Image do autor Leandro Lopes utilizando Ã³culos e sorrindo',
};
