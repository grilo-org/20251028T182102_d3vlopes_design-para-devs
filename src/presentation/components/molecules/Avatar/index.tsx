import { ImageModel } from '@/domain/models/Image';

import { Image } from '@/presentation/components/atoms/Image';

export type AvatarProps = ImageModel;

export const Avatar = ({ src, alt }: AvatarProps) => {
	return (
		<div className="relative bg-background-300 rounded-full w-[11.5625rem] h-[11.5625rem] lg:w-[23.875rem] lg:h-[23.875rem]">
			<Image fill className="rounded-[inherit]" src={src} alt={alt} />
		</div>
	);
};
