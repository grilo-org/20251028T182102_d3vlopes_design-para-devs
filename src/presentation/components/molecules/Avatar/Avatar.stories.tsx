import type { Meta, StoryObj } from '@storybook/react';

import { avatarMock } from './Avatar.mock';

import { Avatar } from '.';

const meta: Meta<typeof Avatar> = {
	title: 'Molecules/Avatar',
	component: Avatar,
	args: avatarMock,
	parameters: {
		backgrounds: {
			default: 'background-400',
		},
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/p1r0o56xHSpG3YHnrNlO3P/LP-Design-para-Devs?type=design&node-id=143-197&mode=design&t=9iKG0T2HeUbm2x4P-0',
		},
	},
};

export default meta;
type Story = StoryObj<typeof Avatar>;

const template: Story = {
	render: (args) => <Avatar {...args} />,
};

export const Default: Story = {
	...template,
};
