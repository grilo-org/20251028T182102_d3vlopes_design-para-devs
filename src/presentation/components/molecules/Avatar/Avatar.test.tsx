import { render, screen } from '@testing-library/react';

import { avatarMock as mock } from './Avatar.mock';

import { Avatar } from '.';

const { getByTestId } = screen;

jest.mock('@/presentation/components/atoms/Image', () => ({
	Image: () => <div data-testid="mock-image" />,
}));

describe('<Avatar />', () => {
	it('should render wrapper with background class', () => {
		const { container } = render(<Avatar {...mock} />);

		const wrapper = container.firstChild;

		expect(wrapper).toHaveClass('bg-background-300');
	});

	it('should render Image component', () => {
		render(<Avatar {...mock} />);

		const image = getByTestId('mock-image');

		expect(image).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<Avatar {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
