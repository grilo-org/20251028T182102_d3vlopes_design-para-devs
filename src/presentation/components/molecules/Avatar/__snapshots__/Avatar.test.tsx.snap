// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should match snapshot 1`] = `
<div>
  <div
    class="relative bg-background-300 rounded-full w-[11.5625rem] h-[11.5625rem] lg:w-[23.875rem] lg:h-[23.875rem]"
  >
    <div
      data-testid="mock-image"
    />
  </div>
</div>
`;
